src/DataAccess/DoctrineMembershipApplicationRepository.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\DataAccess;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\ORMException;
+use WMDE\Fundraising\Entities\MembershipApplication as DoctrineApplication;
+use WMDE\Fundraising\Frontend\Domain\Model\BankData;
+use WMDE\Fundraising\Frontend\Domain\Model\Euro;
+use WMDE\Fundraising\Frontend\Domain\Model\Iban;
+use WMDE\Fundraising\Frontend\Domain\Model\MembershipApplicant;
+use WMDE\Fundraising\Frontend\Domain\Model\MembershipApplication;
+use WMDE\Fundraising\Frontend\Domain\Model\MembershipPayment;
+use WMDE\Fundraising\Frontend\Domain\Model\PersonName;
+use WMDE\Fundraising\Frontend\Domain\Model\PhysicalAddress;
+use WMDE\Fundraising\Frontend\Domain\Repositories\GetMembershipApplicationException;
+use WMDE\Fundraising\Frontend\Domain\Repositories\MembershipApplicationRepository;
+use WMDE\Fundraising\Frontend\Domain\Repositories\StoreMembershipApplicationException;
+
+/**
+ * @license GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class DoctrineMembershipApplicationRepository implements MembershipApplicationRepository {
+
+	private $entityManager;
+
+	public function __construct( EntityManager $entityManager ) {
+		$this->entityManager = $entityManager;
+	}
+
+	public function storeApplication( MembershipApplication $application ) {
+		$doctrineApplication = $this->newDoctrineApplication( $application );
+
+		try {
+			if ( $doctrineApplication->getId() === null ) {
```

Comment:
Is there a particular reason why you set the timestamp manually instead of relying entity mechanisms like in the donation case?
When adding the timestamp on entity creation, we have more consistency across applications.
See https://github.com/wmde/FundraisingStore/blob/master/src/Entities/Donation.php#L131